packages/suite/src/components/suite/SuiteLayout/index.tsx

```diff
@@ -190,25 +191,31 @@ const SuiteLayout = (props: SuiteLayoutProps) => {
 
     // TODO: if (props.layoutSize === 'UNAVAILABLE') return <SmallLayout />;
     const { isMobileLayout, layoutSize } = useLayoutSize();
-    const { guideOpen } = useSelector(state => ({
+    const { guideOpen, isModalOpen } = useSelector(state => ({
         guideOpen: state.guide.open,
+        isModalOpen: state.modal.context !== MODAL.CONTEXT_NONE,
     }));
     const { openGuide, closeGuide } = useActions({
         openGuide: guideActions.open,
         closeGuide: guideActions.close,
     });
+
     useHotkeys(
         'f1,esc',
         e => {
             e.preventDefault();
+            if (e.key === 'Escape' && isModalOpen) {
```

Comment:
It looks like it's not working for 'routed modals'. For example, in settings / device / danger area, it is working for Wipe memory modal, but not for Install firmware modal.

Reply:
a7f27211febf21de3a58d1d926b2dec90cfc6578